cognite/client/_api/datapoints_subscriptions.py

```diff
@@ -258,14 +264,20 @@ def iterate_data(
                 "partitions": [p.dump(camel_case=True) for p in current_partitions],
                 "limit": limit,
                 "pollTimeoutSeconds": poll_timeout,
+                "includeStatus": include_status,
+                "ignoreBadDataPoints": ignore_bad_datapoints,
+                "treatUncertainAsBad": treat_uncertain_as_bad,
             }
             if start is not None:
                 body["initializeCursors"] = start
-            start = None
-
-            res = self._post(url_path=self._RESOURCE_PATH + "/data/list", json=body)
-            batch = _DatapointSubscriptionBatchWithPartitions.load(res.json())
-
+                start = None
+
+            res = self._post(
+                url_path=self._RESOURCE_PATH + "/data/list", json=body, api_subversion=f"{self._api_subversion}-beta"
+            )
```

Comment:
Note to self: wait until GA and remove beta header before merge:
```suggestion
                url_path=self._RESOURCE_PATH + "/data/list", json=body, api_subversion=f"{self._api_subversion}-beta"
            )
```